docs/mmds/mmds-user-guide.md

```diff
@@ -0,0 +1,240 @@
+# microVM Metadata Service
+
+The firecracker microVM Metadata Service (MMDS) is a mutable data store which
+can be used for sharing information between host and guests, in a secure and
+easy at hand way.
+
+# Activating the microVM Metadata Service
+
+By default, MMDS is not reachable from the guest operating system. At microVM
+runtime, MMDS is tightly coupled with a network interface, which allows MMDS
+requests. When configuring the microVM, if MMDS needs to be activated, a
+network interface has to be configured to allow MMDS requests. Network
+interface configuration API can be found in the
+[firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+MMDS_NET_IF=eth0
+curl --unix-socket /tmp/firecracker.socket -i \
+  -X PUT 'http://localhost/network-interfaces/${MMDS_NET_IF}' \
+  -H 'Accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+      "iface_id": "${MMDS_NET_IF}",
+      "guest_mac": "AA:FC:00:00:00:01",
+      "host_dev_name": "tap0"
+      "allow_mmds_requests": true
+    }'
+```
+
+# Configuring the microVM Metadata Service
+
+MMDS can be configured pre-boot only, using the firecracker API server. This
+can be achieved through a HTTP `PUT` request to `/mmds/config` resource. The
+complete MMDS configuration API is described in the [firecracker swagger 
+file](../../src/api_server/swagger/firecracker.yaml).
+
+At the moment, MMDS is configurable with respect to the IPv4 address used by
+guest applications when issuing requests to MMDS. If MMDS configuration is not
+provided before booting up the guest, the MMDS IPv4 address defaults to
+`169.254.169.254`.
+ 
+### Example
+
+```bash
+MMDS_IPV4_ADDR=169.254.170.2
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/mmds/config" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+             "ipv4_address": "${MMDS_IPV4_ADDR}"      
+    }"
+```
+
+MMDS is tightly coupled with a network interface which is used to route MMDS
+packets. To send MMDS intended packets, guest applications must insert a new
+rule into the routing table of the guest OS. This new rule must forward MMDS
+intended packets to a network interface which allows MMDS requests.
+
+### Example
+
+```bash
+MMDS_IPV4_ADDR=169.254.170.2
+MMDS_NET_IF=eth0
+ip route add ${MMDS_IPV4_ADDR} dev ${MMDS_NET_IF}
+```
+
+# Inserting and updating metadata
+
+Inserting and updating metadata is possible through the firecracker API server.
+These operations can be issued before and after the microVM is running. To
```

Comment:
I think worth mention this is done from host side, even though is redundant with the fact that is done by Firecracker API server.
Also please replace firecracker with Firecracker.

Please rephrase These operations can be issued.... to An user can create or update the MMDS datastore before the microVM is started or during its operation.

Reply:
Sure, will rephrase as you suggest.